src/main/java/org/reaktivity/nukleus/http_cache/internal/stream/HttpProxyCacheableRequestGroup.java

```diff
@@ -60,88 +87,218 @@ void setRecentAuthorizationToken(
 
     int getNumberOfRequests()
     {
-        return routeIdsByReplyId.size();
+        MutableInteger totalRequests = new MutableInteger();
+        requestsQueue.forEach((key, routeIdsByReplyId) -> totalRequests.value += routeIdsByReplyId.size());
+        return totalRequests.value;
     }
 
-    boolean queue(
+    void enqueue(
+        String etag,
         long acceptRouteId,
         long acceptReplyId)
     {
+        final boolean requestQueueIsEmpty = requestsQueue.isEmpty();
+        final Long2LongHashMap routeIdsByReplyId = requestsQueue.computeIfAbsent(etag, this::createQueue);
+
         routeIdsByReplyId.put(acceptReplyId, acceptRouteId);
-        return routeIdsByReplyId.size() > 1;
+
+        if (requestQueueIsEmpty)
+        {
+            initiateRequest(etag, acceptRouteId, acceptReplyId);
+        }
+        else if (this.etag != null &&
+                !this.etag.equals(etag))
+        {
+            resetInFightRequest();
+            initiateRequest(null, acceptRouteId, acceptReplyId);
+        }
     }
 
-    void unqueue(
+    void dequeue(
+        String etag,
         long acceptReplyId)
     {
-        routeIdsByReplyId.remove(acceptReplyId);
+        final Long2LongHashMap routeIdsByReplyId = requestsQueue.get(etag);
+        assert routeIdsByReplyId != null;
+
+        final long acceptRouteId = routeIdsByReplyId.remove(acceptReplyId);
+        assert acceptRouteId != routeIdsByReplyId.missingValue();
+        requestsWithAnswer.removeLong(acceptReplyId);
+
         if (routeIdsByReplyId.isEmpty())
         {
-            cleaner.accept(requestHash);
+            requestsQueue.remove(etag);
+
+            if (requestsQueue.isEmpty())
+            {
+                cleaner.accept(requestHash);
+            }
+        }
+
+        if (!requestsQueue.isEmpty())
+        {
+            MutableLong activeRouteId = new MutableLong();
+            MutableLong activeReplyId = new MutableLong();
```

Comment:
If these are only convenience with on stack lifetime, then they can be hoisted to `final` fields and shared instead of recreating each time.

When sharing, remember to reinitialize them to `0L` here to create the same initial state a instantiating new objects each time.